systemd/cgroups.go

```diff
@@ -0,0 +1,126 @@
+package systemd
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/log"
+)
+
+// CPUUsage stores one core's worth of CPU usage for a control group
+// (aka cgroup) of tasks (e.g. both processes and threads).
+// Equivalent to cpuacct.usage_percpu_user and cpuacct.usage_percpu_system
+type CPUUsage struct {
+	CPUId         uint32
+	SystemNanosec uint64
+	UserNanosec   uint64
+}
+
+// CPUAcct stores CPU accounting information (e.g. cpu usage) for a control
+// group (cgroup) of tasks. Equivalent to cpuacct.usage_all
+type CPUAcct struct {
+	CPUs []CPUUsage
+}
+
+// UsageUser returns user (e.g. non-kernel) cpu consumption in nanoseconds, across all available cpu
+// cores, from the point that CPU accounting was enabled for this control group.
+func (c *CPUAcct) UsageUser() (nanoseconds uint64) {
```

Comment:
Works for me. The named return was solely for documentation purposes, since I think we should leave the base unit type in the API somewhere, but I can change this back to `UsageUserNanosecs`. 

I've been consulting the procfs source while making this patch and suspect it was a mistake that they do not include units on their datatypes - I keep having to open the source or visit godocs to remember if something is seconds,jiffies,nanos,etc - so I'm using the prometheus best practise of "have a suffix describing the unit, in plural form" (from https://prometheus.io/docs/practices/naming/#metric-names) in this api